docs/docs/setup-guides/07_configure_the_template.md

```diff
@@ -0,0 +1,89 @@
+---
+sidebar_label: Configure The Template
+sidebar_position: 7
+title: Configure The Template
+---
+import GithubOverrideConfig from '../../static/img/guides/github-override-config.png';
+
+# Flex Project Template Configuration
+
+## The Flex-Config (Front End Configuration)
+
+### hosted configuration
+Generally speaking, when running the *flex-project-template* it loads configuration from the [hosted flex configuration endpoint](https://www.twilio.com/docs/flex/developer/config/flex-configuration-rest-api#ui_attributes). When the *flex-project-template* is deployed, a [`custom_data`](#the-custom_data-object) object is injected into ui_attributes that manages configuration for each feature within the template.
+
+### local configuration
+When running flex locally, the configuration from [hosted flex configuration](https://www.twilio.com/docs/flex/developer/config/flex-configuration-rest-api#ui_attributes) is loaded but the configuration in `plugin-flex-ts-template-v2/public/appConfig.js` overrides anything in hosted configuration. Note the appConfig.js config is only used locally and has no bearing on a deploy or when on hosted flex.
+
+### config management
+
+#### the custom_data object
+The template maintains the configuration that is deployed to [hosted flex configuration](https://www.twilio.com/docs/flex/developer/config/flex-configuration-rest-api#ui_attributes) in version control under the [`flex-config`](/how-it-works/flex-config) folder.  Here you will find `ui_attributes.common.json` file containing the main configuration set.  
+
+At the time of a GitHub action script deploy of the flex template, when an environment is provided, a new file is generated from `ui_attributes.example.json` and it is called `ui_attributes.<env-name>.json`.  After creating this file any placeholder values in the file are replaced as part of the GitHub deployment scripts.  The contents of this file are merged over the top of the `ui_attributes.common.json` file and the output creates the final configuration set that is pushed to the [hosted flex configuration api](https://www.twilio.com/docs/flex/developer/config/flex-configuration-rest-api#ui_attributes).
+
+#### the data model
+
+The custom_data model that is pushed into ui_attributes follows this pattern
```

Comment:
I think we should mention that we replace hyphens in the feature name with underscores for config.